scipio/src/sys/posix_buffers.rs

```diff
@@ -82,25 +68,23 @@ impl PosixDmaBuffer {
         }
         let len = std::cmp::min(dst.len(), self.size - offset);
 
-        let dst_ptr = dst.as_mut_ptr() as *mut u8;
-        assert_eq!(is_nonoverlapping(self.as_ptr(), dst_ptr, len), true);
+        let dst_ptr = dst.as_mut_ptr();
         unsafe { std::ptr::copy_nonoverlapping(self.as_ptr().add(offset), dst_ptr, len) }
         len
     }
 
-    pub fn copy_from_slice(&self, offset: usize, src: &[u8]) -> usize {
+    pub fn copy_from_slice(&mut self, offset: usize, src: &[u8]) -> usize {
         if offset > self.size {
             return 0;
         }
         let len = std::cmp::min(src.len(), self.size - offset);
 
-        let src_ptr = src.as_ptr() as *const u8;
-        assert_eq!(is_nonoverlapping(src_ptr, self.as_ptr(), len), true);
+        let src_ptr = src.as_ptr();
         unsafe { std::ptr::copy_nonoverlapping(src_ptr, self.as_mut_ptr().add(offset), len) }
         len
     }
 
-    pub fn memset(&self, value: u8) {
+    pub fn memset(&mut self, value: u8) {
```

Comment:
@matklad this is pre-existing of course, but should have size asserts here, no ?

I am inclined towards merging this now even if we can simplify later because it provides a very clear benefit.

Reply:
Humm, I guess this one can live without the assert because the size is not passed by the user.

Reply:
Exactly!

Reply:
As for the simplification, I suspect we can do two things:

* just use global allocator instead of megalign: modern std api allow to specify alignment (no link: I am on mobile)
* reduce unsafe by leveraging [u8]::copy_from_slce

Reply:
I sent a PR today to deal with your second suggestion. But look at it later, go enjoy your life ! Don't text and drink.

About the allocator, I am less hot on it: With seastar with had major issues with locks inside the allocator to the point we had to have our own.

Until I understand the real life issues of this better for rust I wasn't planning on doing any serious work around allocation.